### PR TITLE
Upgrading iTwin.js packages to 4.0

### PR DESCRIPTION
**Pull request by Jake-Screen**
upgraded iTwin.js packages to 4.0 and fixed removed imports

### CODE REVIEW

File: typescript/integration-tests/package.json
Comment (aruniverse) [inline typescript/integration-tests/package.json:26]: update the reality-data-client across the monorepo to 1.0.0

File: typescript/sdk-node/package.json
Comment (aruniverse) [inline typescript/sdk-node/package.json:46]: 4.0 only support node 18+, so update that across the board. pkgs, pipelines, etc

File: typescript/integration-tests/package.json
Comment (Jake-Screen) [reply typescript/integration-tests/package.json]: updated reality-data-client and service-authorization

File: typescript/sdk-node/package.json
Comment (Jake-Screen) [reply typescript/sdk-node/package.json]: updated

Comment (aruniverse) [reply typescript/sdk-node/package.json]: 18.x will cause segfaults as mentioned in scrum yesterday

Comment (Jake-Screen) [reply typescript/sdk-node/package.json]: fixed similar to link from issue post

**Review by aruniverse — APPROVED** [review]
lgtm; going to defer to denis to run locally and approve before merging this pr

**Review by dbiguenet — APPROVED** [review]

**Pull request merged**
